### ISSUE TITLE
Calling "TableCell.setDataStyle()" before setting the cell value has no effect

### ISSUE THREAD

**Issue by negora** [jferard/fastods#140]
I'm not sure whether this issue is really a bug or not. Apologizes in advance if it isn't...

If I call `TableCell.setDataStyle()` **before** setting the cell value (with `TableCell.setFloatValue()` for example), then the style is not applied at all to the cell. Is it the expected behaviour or is it a "bug"? 

Thank you!

**Comment by jferard** [comment]
This one is harder. I didn't notice that behaviour. 
It seems a good idea, in theory, to let the user set the style before the content, but I have to think about it a little more.

If I decide this is a feature and not a bug, you'll have a full explanation!

**Comment by negora** [comment]
@jferard Perfect. Thank you for investigating this.

**Comment by jferard** [comment]
TLDR; I think that's a bug!

Here's the current politic. When you set the value of a cell, e.g.:

    cell.setFloatValue(10.5);
    
FastODS:

* sets the value and the type of the cell;
* if there is **no** current data-style in cell-style, row-style or column-style, takes the default one from `DataStyles`;
* if there is **a** current data-style in cell-style, row-style or column-style, replaces this data-style with the default one from `DataStyles`.

(There's a complex merge between cell styles and cell data styles here, but it's not relevant for this issue).

Hence, if you have set the data style of the row, it will be overriden by the default data style.

I need to do some testing before I fix it (I was afraid of the performance impact, but it seems that's will be ok.)

**Comment by negora** [comment]
> if there is a current data-style in cell-style, row-style or column-style, replaces this data-style with the default one from DataStyles.

I noticed that behaviour on my first trials with FastODS . But I wasn't sure whether that problem was connected to the one that I mention in this ticket . I finally ended up applying the data styles separately from the cell styles, to solve that.

**Comment by jferard** [comment]
@negora Can you try the last version?

**Comment by negora** [comment]
@jferard Tested and... Works OK :) . Thank you!

**Comment by jferard** [comment]
Closing. Feel free to re-open if necessary.

**Comment by jferard** [comment]
Closing. Feel free to re-open if necessary.
